### PR TITLE
Add support for enabling extensions via SDL

### PR DESCRIPTION
**Pull request by elprans**
The new `using extension <extname> [version 'ver'];` syntax is the 
equivalent for `CREATE EXTENSION <extname>;` DDL.  We are stepping away 
from the usual custom of simply stripping the `CREATE` verb, because the 
resulting `extension <foo>;` is a bit _too_ terse and might at a glance be
confused with a declaration that puts the entire module into an extension.

### CODE REVIEW

File: edb/edgeql-parser/src/keywords.rs [hunk 1]
```diff
@@ -45,7 +43,6 @@ pub const UNRESERVED_KEYWORDS: &[&str] = &[
     "named",
     "object",
     "of",
-    "oids",
```
Comment (1st1) [inline edb/edgeql-parser/src/keywords.rs:48]: Thanks God.

File: tests/test_edgeql_syntax.py [hunk 2]
```diff
@@ -3057,6 +3057,31 @@ def test_edgeql_syntax_ddl_create_migration_06(self):
...
+
+    def test_edgeql_syntax_ddl_create_migration_08(self):
+        """
+        START MIGRATION TO {
+            using extension graphql;
+        };
+        """
+
+    @tb.must_fail(errors.EdgeQLSyntaxError,
+                  "'using extension' cannot be used inside a module block")
```
Comment (1st1) [inline tests/test_edgeql_syntax.py:3075]: I usually add col/line checks for this kind of tests.

**Review by 1st1 — APPROVED** [review]

File: tests/test_edgeql_data_migration.py [hunk 3]
```diff
@@ -7714,6 +7714,88 @@ async def test_edgeql_migration_non_ddl_statements(self):
...
+                }
+                FILTER .name = 'graphql'
+            """,
+            [{
+                'name': 'graphql',
+            }]
+        )
+
+        await self.con.execute('''
+            START MIGRATION TO {
```
Comment (dmgolembiowski) [inline tests/test_edgeql_data_migration.py:7751]: Is this migration committing changes to the default module?

Comment (elprans) [reply tests/test_edgeql_data_migration.py:7751]: No.  Extensions are database-wide, and so are declared at the top level.

**Comment by dmgolembiowski** [comment]
>> `map` container will be removed from EdgeDB.
>> 
>> We can have a `map` if and when we will implement it natively as a PostgreSQL type extension.
> Comment originally by @vpetrovykh https://github.com/edgedb/edgedb/issues/192#issue-330081977

Alright, please hold my beer.

**Pull request merged**